### PR TITLE
[Buildkite] Fix logic to check which packages are excluded in Kibana (Serverless)

### PR DESCRIPTION
**Pull request by mrodm**
## Proposed commit message

~Add support to exclude some packages by defining a new environment variable.~

~There are some packages that are trying to be installed in a Observability project in Elastic Serverless, but they cannot be installed.~

~Example of build: https://buildkite.com/elastic/integrations-serverless/builds/400~

Updated Buildkite scripts to check which packages are excluded in the current Kibana configuration of each Serverless project type.

Here in this build it is shown that `universal_profiling_agent` (profiler_agent) package is skipped because it is added into the excluded list of Kibana.

### CODE REVIEW

**Review by jlind23 — APPROVED** [review]

**Comment by andrewkroh** [comment]
> but they cannot be installed

Why can they not be installed? Is this a temporary limitation? Do these packages need to define a `capabilities` value in their manifest?

**Comment by mrodm** [comment]
> > but they cannot be installed
> 
> Why can they not be installed? Is this a temporary limitation? Do these packages need to define a `capabilities` value in their manifest?

This is the error for `security_detection_engine` package:
> Error: error running package asset tests: could not complete test run: can't install the package: there was an apply error: installation failed: can't install the package: could not zip-install package; API status code = 400; response body = {"statusCode":400,"error":"Bad Request","message":"Encountered 5224 errors creating saved objects: ...

This is the error for `universal_profile_agent` package:
>  Error: can't install the package: could not zip-install package; API status code = 403; response body = {"statusCode":403,"error":"Forbidden","message":"profiler_agent installation is not authorized"}

Probably `security_detection_engine` would require to define capabilities in its manifest, but there are already quite a few versions published with spec version >= 3.0.0. Those versions would be available in Observability projects too.

About `universal_profiling_agent` not sure what would be needed, but currently that package cannot be installed in Observability projects. In Security projects it works. This package has also a couple of versions using format_version >= 3.0.0.

Do you know if it can be done something else for these packages ? @jsoriano 
Should `universal_profiling_agent` be available in Observability projects too ?

**Comment by jsoriano** [comment]
> This is the error for `security_detection_engine` package:
> 
> > Error: error running package asset tests: could not complete test run: can't install the package: there was an apply error: installation failed: can't install the package: could not zip-install package; API status code = 400; response body = {"statusCode":400,"error":"Bad Request","message":"Encountered 5224 errors creating saved objects: ...

> Probably security_detection_engine would require to define capabilities in its manifest, but there are already quite a few versions published with spec version >= 3.0.0. Those versions would be available in Observability projects too.

Do you have the full error message? This is likely complaining about an unsupported saved object type. This package should probably define the `security` capability.

For packages already published, we may need to end up removing packages from the registry if they cause problems. In any case this package should be included in the exclude list for observability till these issues are handled https://github.com/elastic/kibana/blob/6c7907a2b7aa907ab06db758a7bf01c6c65290aa/config/serverless.oblt.yml#L52

> This is the error for `universal_profile_agent` package:
> 
> > Error: can't install the package: could not zip-install package; API status code = 403; response body = {"statusCode":403,"error":"Forbidden","message":"profiler_agent installation is not authorized"} 

> About `universal_profiling_agent` not sure what would be needed, but currently that package cannot be installed in Observability projects. In Security projects it works. This package has also a couple of versions using format_version >= 3.0.0.

No idea where this error comes from, but looks like this package should also include the security capability. And might be included in the observability exclude list in the meantime.

**Comment by mrodm** [comment]
> Do you have the full error message? This is likely complaining about an unsupported saved object type. This package should probably define the `security` capability.

The error message is too long. This is the first saved objects failing:

```
Error: error running package asset tests: could not complete test run: can't install the package: there was an apply error: installation failed: can't install the package: could not zip-install package; API status code = 400; response body = {"statusCode":400,"error":"Bad Request","message":"Encountered 5224 errors creating saved objects: [{\"type\":\"security-rule\",\"id\":\"000047bb-b27a-47ec-8b62-ef1a5d2c9e19_102\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"000047bb-b27a-47ec-8b62-ef1a5d2c9e19_103\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"000047bb-b27a-47ec-8b62-ef1a5d2c9e19_104\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"000047bb-b27a-47ec-8b62-ef1a5d2c9e19_105\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"000047bb-b27a-47ec-8b62-ef1a5d2c9e19_106\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"000047bb-b27a-47ec-8b62-ef1a5d2c9e19_207\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_105\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_106\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_107\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_108\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_109\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_110\",\"error\":{\"type\":\"unsupported_type\"}},{\"type\":\"security-rule\",\"id\":\"00140285-b827-4aee-aa09-8113f58a08f3_111\",\"error\":{\"type\":\"unsupported_type\"}}...
```

Full log can be downloaded from the buildkite step: https://buildkite.com/elastic/integrations-serverless/builds/407#018f74bb-d02c-45e1-95bf-c865ae95f05b

**Comment by mrodm** [comment]
About `universal_profiling_agent` package probably should be available in Observability ? 
This is the documentation about it: https://www.elastic.co/guide/en/observability/current/universal-profiling.html

**Comment by jsoriano** [comment]
> About `universal_profiling_agent` package probably should be available in Observability ? This is the documentation about it: https://www.elastic.co/guide/en/observability/current/universal-profiling.html

Ah yes, you are right, profiling should work in observability. @elastic/profiling do you know what could be causing the failure mentioned in https://github.com/elastic/integrations/pull/9849#issuecomment-2109621665: `profiler_agent installation is not authorized`.

**Comment by mrodm** [comment]
> > About `universal_profiling_agent` package probably should be available in Observability ? This is the documentation about it: https://www.elastic.co/guide/en/observability/current/universal-profiling.html
> 
> Ah yes, you are right, profiling should work in observability. @elastic/profiling do you know what could be causing the failure mentioned in [#9849 (comment)](https://github.com/elastic/integrations/pull/9849#issuecomment-2109621665): `profiler_agent installation is not authorized`.

@elastic/profiling should the `universal_profiling_agent` package be available for observability projects in Elastic Serverless? Currently, this package fails with this error:

> Error: can't install the package: could not zip-install package; API status code = 403; response body = {"statusCode":403,"error":"Forbidden","message":"profiler_agent installation is not authorized"}

If so, I could update the list of packages to be skipped to just "security_detection_engine" package for now.

**Comment by mrodm** [comment]
`univeral_profiliing_agent` should be skipped according to the kibana config, it has been fixed here: https://github.com/elastic/integrations/pull/9849/commits/f4f41bd66a2f057fb67e7abe4a117c36cbb5dd09

In this test build, it is already shown as skipped: https://buildkite.com/elastic/integrations-serverless/builds/425

> For packages already published, we may need to end up removing packages from the registry if they cause problems. In any case this package should be included in the exclude list for observability till these issues are handled https://github.com/elastic/kibana/blob/6c7907a2b7aa907ab06db758a7bf01c6c65290aa/config/serverless.oblt.yml#L52

According to this, I think it would be better to create a PR to update the exclude list to add `security_detection_engine` there. WDYT? @kpollich @jsoriano 

Added `security` capability in security_detection_engine here: https://github.com/elastic/integrations/pull/9938

File: .buildkite/scripts/common.sh [hunk 1]
```diff
@@ -408,7 +412,10 @@ is_package_excluded() {
     if [[ "${excluded_packages}" == "null" ]]; then
         return 1
     fi
-    if echo "${excluded_packages}" | grep -q -E "\"${package}\""; then
+    local package_name=""
+    package_name=$(package_name_manifest)
+
+    if echo "${excluded_packages}" | grep -q -E "\"${package_name}\""; then
```
Comment (mrodm) [inline .buildkite/scripts/common.sh:418]: Retrieved the package name from the manifest to compare with the values from Kibana.

**Review by jsoriano — APPROVED** [review]

**Comment by mrodm** [comment]
CI failures are unrelated to this change, merging it.

**Pull request merged**